### PR TITLE
Build: Add `punycode` to transpiled dependencies

### PR DESCRIPTION
**Pull request by ockham**
_Merge #33492 and #33496 and rebase this PR before reviewing!_

#### Changes proposed in this Pull Request

Build: Add `punycode` to transpiled dependencies

Part 3 in a series of fixes for for https://github.com/Automattic/jetpack/issues/12511. Quoting part 1 (#33492):

> I would like to move [...] logic that's taking care of transpiling certain `node_modules` into the `calypso-build` package. The reason for this is that Jetpack (which uses `calypso-build` in its toolchain) has currently [a number of Gutenberg blocks that are broken in IE11](https://github.com/Automattic/jetpack/issues/12511) because of a 3rd party dependency that ships untranspiled code. See https://github.com/Automattic/jetpack/issues/12511#issuecomment-497593885 for how I would like to fix this.

#### Testing instructions

```
npm run distclean && npm ci
npm start
```

Verify that the app still builds and runs. In particular, check sections that involve any of the transpiled dependencies (Stats screen for `d3`) still work.

_Equally important:_ `npm link` Jetpack against the `@automattic/calypso-build` package on this branch, and rebuild it. Test it in IE11 to make sure https://github.com/Automattic/jetpack/issues/12511 is fixed.

#### Follow-up

Release the new version, and file a Jetpack PR to use it.

### CODE REVIEW

File: packages/calypso-build/package.json
Comment (ockham) [inline packages/calypso-build/package.json]: https://semver.org/ says:

> Given a version number MAJOR.MINOR.PATCH, increment the:
>
> 1. MAJOR version when you make incompatible API changes,
> 2. MINOR version when you add functionality in a backwards-compatible manner, and
> 3. PATCH version when you make backwards-compatible bug fixes.

We're not touching the API here, and I think we are backwards-compat (worst thing that can happen is double transpilation of some stuff?)

**Review by sirreal — APPROVED** [review]
Pending other PRs 👍

**Comment by ockham** [comment]
Trying to build Jetpack (`yarn build`) while `npm link`ing against `@automattic/calypso-build` on this branch is currently failing for me with 

```
  Error: ENOENT: no such file or directory, lstat '~/src/jetpack/node_modules/@automattic/calypso-build/node_modules/chalk/index.js.flow'
```

:slightly_frowning_face:

**Comment by ockham** [comment]
Obsoleted by #38531.

**Comment by jsnajdr** [comment]
@ockham After #38531, Punycode is still not in the transpilation whitelist. Should it be added?

**Comment by sirreal** [comment]
> @ockham After #38531, Punycode is still not in the transpilation whitelist. Should it be added?

Yes it should. It has a `module` field that points at this file which appears to require transpilation:

https://unpkg.com/browse/punycode@2.1.1/punycode.es6.js

**Comment by jsnajdr** [comment]
> Yes it should.

OK then let's do a quick PR before publishing `@automattic/calypso-build@5.2.0`. I'll wait for @ockham's #38862 to get merged to prevent Changelog conflicts.

**Comment by ockham** [comment]
Ah, my bad. I was in a bit of a PR closing spree :grimacing:

**Comment by ockham** [comment]
> > Yes it should.
> 
> OK then let's do a quick PR before publishing `@automattic/calypso-build@5.2.0`. I'll wait for @ockham's #38862 to get merged to prevent Changelog conflicts.

Merged #38862. Do feel free to merge my approved PRs on my behalf though if they'd otherwise block you (especially if it's just doc fixes) :smile:

**Pull request closed without merging**
